deployment/deploymentClass.js

```diff
@@ -166,34 +170,35 @@ class Deployment {
         return !isNaN(element);
       })
     ) {
-      throw "See deployment.json: (2) version is not valid - must be in format x.x.x (e.g. 1.3.1)";
+      throw "See deployment.json: (2) version is not valid - make sure to use integers between the periods - must be in format x.x.x (e.g. 1.3.1)";
     }
   }
 
   // Checks if the protocol level data in the deployment json file is present and/or valid.
   checkProtocolLevelData(protocol) {
-    if (this.data[protocol] == undefined) {
+    if ([undefined, ""].includes(this.data[protocol])) {
```

Comment:
You can do `if (!this.data[protocol])`.

Both `undefined` and `""` evaluate to false.

Reply:
Actually, when I have tried this. And after testing again. "" does not evaluate as false with this expression.

Reply:
How strange 🤔 
Being how JS is I believe anything 😅 

Theres this thing of using `!!` when you want to evaluate some non-boolean as a boolean. And `===` whenever you need to compare, to make sure you are always comparing what you intend to compare.

Let's leave it as it is then.

<img width="518" alt="Screenshot 2022-09-02 at 17 55 44" src="https://user-images.githubusercontent.com/17258946/188192693-5bfdd319-0f0c-4578-ac72-6f0f270b7c1a.png">